pytorch_lightning/plugins/environments/cluster_environment.py

```diff
@@ -12,23 +12,30 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 from abc import ABC, abstractmethod
+from typing import Any, Type
 
 
 class ClusterEnvironment(ABC):
     """Specification of a cluster environment."""
 
+    def __new__(cls, *args: Any, **kwargs: Any) -> "ClusterEnvironment":
+        # TODO: remove in 1.7
+        _check_for_deprecated_methods(cls)
```

Comment:
just check? not filling the old master_ to new main_

Reply:
I think it can't be done so easily, because one is a property and the other is a method. I'm not sure what the best is actually. Since the new properties are also abstract, the user is forced to implement them. So a deprecation message is maybe not appropriate. Should it be a direct error?

Note: This only concerns people implementing a custom cluster plugin. Anyone using an existing one is fine.

Reply:
Neat ! I like this design. Might be worth to automate this for most classes :)